app/gui/src/controller/graph/executed.rs

```diff
@@ -228,19 +228,26 @@ impl Handle {
         if self.project.read_only() {
             Err(ReadOnly.into())
         } else {
-            for local_call in stack.clone() {
+            let mut successful_calls = Vec::new();
+            let mut result = Ok(());
+            for local_call in stack {
                 debug!("Entering node {}.", local_call.call);
-                self.execution_ctx.push(local_call).await?;
+                if let Err(error) = self.execution_ctx.push(local_call.clone()).await {
+                    result = Err(error);
+                    break;
+                }
+                successful_calls.push(local_call);
```

Comment:
Here as well, `take_while(...).collect_vec()`may be a bit more readable as we can make one variable immutable + one early return from loop is removed.